glogger/messages_buffer.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+from typing import List
+
+
+class MessagesBuffer:
+    """
+    A list of strings limited by the total length of all items.
+    Keeps count of current number of items, and dropped and added items.
+    Not thread-safe on its own!
+    """
+
+    def __init__(self, max_total_length: int, overflow_drop_factor: float):
+        assert max_total_length > 0, "max_total_length must be positive!"
+        self.max_total_length = max_total_length  # maximum size of buffer in bytes
+        self.overflow_drop_factor = overflow_drop_factor  # drop this percentage of messages upon overflow
+        self.total_length = 0
+        self.buffer: List[str] = []
+        self.lengths: List[int] = []
+        self.head_serial_no = 0
+        self.dropped = 0
+
+    @property
+    def count(self) -> int:
+        """Number of strings currently in the buffer."""
+        return len(self.buffer)
+
+    @property
+    def utilized(self) -> float:
+        """Total length used divided by maximum total length."""
+        return self.total_length / self.max_total_length
+
+    @property
+    def next_serial_no(self) -> int:
+        """The serial number of the next item to be inserted."""
+        return self.head_serial_no + self.count
```

Comment:
Possible race condition? If head_serial_no changes but count doesnt? Or is it atomic?